my.component.ts

```diff
@@ -0,0 +1,28 @@
+import { Component, OnInit } from '@angular/core';
+import { ApiService } from './api.service';
+import { ToasterService } from './ToasterService';
+
+@Component({
+  selector: 'app-my-component',
+  templateUrl: './my-component.component.html',
+  styleUrls: ['./my-component.component.css']
+})
+export class MyComponent implements OnInit {
+
+  students: any;
+  subjects: any;
+
+  constructor(private apiService: ApiService, private toasterService: ToasterService) { }
+
+  ngOnInit() {
+    this.apiService.getStudents().subscribe(
+      response => this.students = response,
+      error => this.toasterService.error(error)
+    );
```

Comment:
**Avoid use of multiple "subscribe" calls and implement "forkJoin" while calling multiple APIs in parallel**

> To optimize the provided code, you can use "forkJoin" to combine multiple HTTP requests and ensure better performance and code readability. Optimization can be achieved with forkJoin like single subscription, parallel requests, consistency and readability, error handling.
> - forkJoin triggers both getStudents() and getSubjects() simultaneously, resulting in better performance by fetching data in parallel.
> - By combining requests into one observable, you ensure that either both sets of data are received successfully, or an error is caught for both, maintaining data consistency.

```
const request1$ = this.apiService.getStudents();
const request2$ = this.apiService.getSubjects();
forkJoin([request1$, request2$]).subscribe(([data1, data2]) => {
  this.students = data1;
  this.subjects = data2;
}, error => this.toasterService.error(error));
```